keras_cv/layers/preprocessing/rand_augment.py

```diff
@@ -0,0 +1,252 @@
+            self.translate_x,
+            self.translate_y,
+            self.cutout,
+        ]
+
+    @tf.function
+    def augment_sample(self, sample):
+        for _ in range(self.num_layers):
+            selected_op = tf.random.uniform(
+                (), maxval=len(self.augmentation_layers) + 1, dtype=tf.int32
+            )
+            branch_fns = []
+            for (i, layer) in enumerate(self.augmentation_layers):
+                branch_fns.append((i, lambda: layer(sample)))
+
+            augmented_sample = tf.switch_case(
+                branch_index=selected_op, branch_fns=branch_fns, default=lambda: sample
+            )
+            if self.probability_to_apply is not None:
+                augmented_sample = tf.cond(
+                    tf.random.uniform(shape=(), dtype=tf.float32)
+                    < self.probability_to_apply,
+                    lambda: augmented_sample,
+                    lambda: sample,
+                )
+            sample = augmented_sample
+        return sample
+
+    def call(self, inputs):
+        inputs["images"] = preprocessing_utils.transform_value_range(
+            inputs["images"], self.value_range, (0, 255)
+        )
+        result = tf.map_fn(lambda sample: self.augment_sample(sample), inputs)
+        result["images"] = preprocessing_utils.transform_value_range(
+            result["images"], (0, 255), self.value_range
+        )
+        return result
+
+    def get_config(self):
+        return {
+            "num_layers": self.num_layers,
+            "magnitude": self.magnitude,
+            "magnitude_standard_deviation": self.magnitude_standard_deviation,
+            "probability_to_apply": self.probability_to_apply,
+            "value_range": self.value_range,
+        }
+
+
+def auto_contrast_policy(magnitude, magnitude_std):
+    return {}
+
+
+def equalize_policy(magnitude, magnitude_std):
+    return {}
+
+
+def solarize_policy(magnitude, magnitude_std):
+    # TODO(lukewood): this should support a sample-able factor.
+    return {"threshold": magnitude / 10 * 256}
+
+
+def solarize_add_policy(magnitude, magnitude_std):
+    # TODO(lukewood): this should support a sample-able factor.
+    return {"addition": magnitude / 10 * 110, "threshold": 128}
+
+
+def invert_policy(magnitude, magnitude_std):
+    return {"addition": 0, "threshold": 0}
+
+
+def color_policy(magnitude, magnitude_std):
+    factor = core.NormalFactor(
+        mean=magnitude / 10.0,
+        standard_deviation=magnitude_std / 10.0,
+        min_value=0,
+        max_value=1,
+    )
+    return {"factor": factor}
+
+
+def contrast_policy(magnitude, magnitude_std):
+    # TODO(lukewood): should we integrate RandomContrast with `factor`?
+    # RandomContrast layer errors when factor=0
+    factor = max(magnitude / 10, 0.001)
+    return {"factor": factor}
+
+
+def brightness_policy(magnitude, magnitude_std):
+    # TODO(lukewood): should we integrate RandomBrightness with `factor`?
```

Comment:
This is in core keras, unfortunately it can't take factor object. maybe core keras can somehow support this?